yoti-sdk-api/src/main/java/com/yoti/api/client/spi/remote/call/identity/IdentityService.java

```diff
@@ -0,0 +1,96 @@
+package com.yoti.api.client.spi.remote.call.identity;
+
+import static com.yoti.api.client.spi.remote.call.HttpMethod.HTTP_POST;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.AUTH_ID_HEADER;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.CONTENT_TYPE;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.CONTENT_TYPE_JSON;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.DEFAULT_IDENTITY_URL;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.PROPERTY_YOTI_API_URL;
+import static com.yoti.api.client.spi.remote.util.Validation.notNull;
+import static com.yoti.api.client.spi.remote.util.Validation.notNullOrEmpty;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+
+import com.yoti.api.client.identity.ShareSession;
+import com.yoti.api.client.identity.ShareSessionRequest;
+import com.yoti.api.client.spi.remote.call.ResourceException;
+import com.yoti.api.client.spi.remote.call.SignedRequest;
+import com.yoti.api.client.spi.remote.call.SignedRequestBuilderFactory;
+import com.yoti.api.client.spi.remote.call.factory.UnsignedPathFactory;
+import com.yoti.json.ResourceMapper;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IdentityService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(IdentityService.class);
+
+    private final UnsignedPathFactory pathFactory;
+    private final SignedRequestBuilderFactory requestBuilderFactory;
+
+    private final String apiUrl;
+
+    public IdentityService(UnsignedPathFactory pathFactory, SignedRequestBuilderFactory requestBuilderFactory) {
+        this.pathFactory = pathFactory;
+        this.requestBuilderFactory = requestBuilderFactory;
+
+        this.apiUrl = System.getProperty(PROPERTY_YOTI_API_URL, DEFAULT_IDENTITY_URL);
+    }
+
+    public static IdentityService newInstance() {
+        return new IdentityService(new UnsignedPathFactory(), new SignedRequestBuilderFactory());
+    }
+
+    public ShareSession createShareSession(String appId, KeyPair keyPair, ShareSessionRequest shareSessionRequest)
+            throws IdentityException {
+        notNullOrEmpty(appId, "Application id");
+        notNull(keyPair, "Application key Pair");
+        notNull(shareSessionRequest, "Share Session request");
+
+        LOG.info("Requesting Share Session Creation for SDK ID '{}", appId);
```

Comment:
This still is going to be "spam" for a RP as it's very likely that they will always use the same App ID (so every log message is going to look the same)... It's better than no dynamic info but I feel like it should probably be a `debug` or even `trace` log level message.